packages/cli/generation/ir-generator/src/dynamic-snippets/DynamicSnippetsConverter.ts

```diff
@@ -140,7 +140,17 @@ export class DynamicSnippetsConverter {
             declaration: this.convertDeclaration({ name: endpoint.name, fernFilepath: endpoint.fernFilepath }),
             location,
             request: this.convertRequest({ endpoint }),
-            response: DynamicSnippets.Response.json(),
+            response: (endpoint.response?.body?._visit<DynamicSnippets.Response>({
+                json: () => DynamicSnippets.Response.json(),
+                streaming: () => DynamicSnippets.Response.streaming(),
+                streamParameter: () => DynamicSnippets.Response.streamParameter(),
+                fileDownload: () => DynamicSnippets.Response.fileDownload(),
+                text: () => DynamicSnippets.Response.text(),
+                bytes: () => DynamicSnippets.Response.bytes(),
+                _other: () => DynamicSnippets.Response.json()
+            }) ?? DynamicSnippets.Response.json()) as DynamicSnippets.Response.Json,
```

Comment:
why do we have the cast here? I get why we have the default but not the cast :)

Reply:
The cast isn't necessary, dropping